pybricksdev/connections/pybricks.py

```diff
@@ -175,6 +176,19 @@ def pybricks_service_handler(self, _: int, data: bytes) -> None:
             (flags,) = struct.unpack_from("<I", data, 1)
             self.status_observable.on_next(StatusFlag(flags))
 
+    @asynccontextmanager
+    async def connection(self, device: BLEDevice):
+        """Context manager to handle connect and disconnect for a device discovered
+        with :meth:`pybricksdev.ble.find_device`. To be used in an ``async with``
+        statement.
+
+        Args:
+            device: The device to connect to.
+        """
+        await self.connect(device)
+        yield
+        await self.disconnect()
```

Comment:
Does this need to be:
```suggestion
        await self.connect(device)
        try:
            yield
        finally:
            await self.disconnect()
```

...to ensure that disconnect is always called even if the code in the `async with` block throws an exception?

Reply:
Yeah you're right. I didn't know how this decorator worked exactly.